{
    "source": "BDSA",
    "name": "BDSA-2018-2174",
    "title": "Linux Kernel Vulnerable to Denial-of-Service via Integer Overflow in  'posix-timers.c'",
    "description": "Linux kernel contains a denial-of-service (DoS) vulnerability.  This allows a local attacker to crash a system via crafted `mmap`, `futex`, `timer_create`, and `timer_settime` system calls.",
    "technicalDescription": "The DoS occurs due to an Integer overflow in `kernel/time/posix-timers.c`.  The overrun value for timing discrepancy handling is declared as an integer value, although the forwarding values can return a sum value larger than `2147483647`, which is `MAX_INT`.  When this value is retrieved using standard functions such as `timer_getoverrun` and `siginfo::si_overrun` a almost random number will be returned.  This will cause multiple kernel issues depending on the value.",
    "publishedDate": "2018-08-02T15:07:01.606Z",
    "updatedDate": "2018-08-02T15:07:01.597Z",
    "disclosureDate": "2018-06-22T00:00:00.000Z",
    "exploitPublishDate": "2018-06-25T00:00:00.000Z",
    "solution": "",
    "severity": "MEDIUM",
    "cvss2": {
        "baseScore": 4.9,
        "impactSubscore": 6.9,
        "exploitabilitySubscore": 3.9,
        "severity": "MEDIUM",
        "accessVector": "LOCAL",
        "accessComplexity": "LOW",
        "authentication": "NONE",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "COMPLETE",
        "temporalMetrics": {
            "exploitability": "PROOF_OF_CONCEPT",
            "remediationLevel": "WORKAROUND",
            "reportConfidence": "CONFIRMED",
            "score": 4.2
        },
        "vector": "(AV:L/AC:L/Au:N/C:N/I:N/A:C/E:POC/RL:W/RC:C)"
    },
    "cvss3": {
        "baseScore": 6.2,
        "impactSubscore": 3.6,
        "exploitabilitySubscore": 2.5,
        "severity": "MEDIUM",
        "attackVector": "LOCAL",
        "attackComplexity": "LOW",
        "confidentialityImpact": "NONE",
        "integrityImpact": "NONE",
        "availabilityImpact": "HIGH",
        "privilegesRequired": "NONE",
        "scope": "UNCHANGED",
        "userInteraction": "NONE",
        "temporalMetrics": {
            "exploitability": "PROOF_OF_CONCEPT",
            "remediationLevel": "WORKAROUND",
            "reportConfidence": "CONFIRMED",
            "score": 5.7
        },
        "vector": "CVSS:3.0/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H/E:P/RL:W/RC:C"
    },
    "useCvss3": true,
    "classifications": [],
    "zeroDay": false,
    "underReview": true,
    "parentAdvisory": false,
    "credit": "Team OWL337",
    "workaround": "A third party patch is currently available [here](https://github.com/torvalds/linux/commit/78c9c4dfbf8c04883941445a195276bb4bb92c76).",
    "bdsaTags": [],
    "overallScore": 5.7,
    "_meta": {
        "allow": [],
        "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/BDSA-2018-2174",
        "links": [
            {
                "rel": "cwes",
                "href": "https://trendmicro.app.blackduck.com/api/cwes/CWE-136"
            },
            {
                "rel": "related-vulnerability",
                "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/CVE-2018-12896",
                "label": "NVD"
            },
            {
                "rel": "bdsa-ranges",
                "href": "https://trendmicro.app.blackduck.com/api/vulnerabilities/BDSA-2018-2174/ranges"
            },
            {
                "rel": "reference",
                "href": "https://bugzilla.kernel.org/show_bug.cgi?id=200189",
                "label": "ADVISORY"
            },
            {
                "rel": "reference",
                "href": "https://github.com/lcytxw/bug_repro/tree/master/bug_200189",
                "label": "POC"
            },
            {
                "rel": "reference",
                "href": "https://github.com/torvalds/linux/commit/78c9c4dfbf8c04883941445a195276bb4bb92c76",
                "label": "PATCH"
            }
        ]
    }
}